[Operator Versioning][Edge] Have operator version number available at the loading stage

cccclai opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67731
* #67728
* #67730
* __->__ #67729



1. operator version is needed to decide whether applying upgrader or not. This pr make it available at loading stage.
2. Swap the order of parsing instruction and operator, because instruction needs to know the operator first because deciding whether applying upgrader or not (change `OP` to `CALL` or not).

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516/)

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516)

Commit:
[Operator Versioning][Edge] Have operator version number available at the loading stage

1. operator version is needed to decide whether applying upgrader or not. This pr make it available at loading stage.
2. Swap the order of parsing instruction and operator, because instruction needs to know the operator first because deciding whether applying upgrader or not (change `OP` to `CALL` or not).

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516/)

[ghstack-poisoned]

Commit:
Update on "[Operator Versioning][Edge] Have operator version number available at the loading stage"


1. operator version is needed to decide whether applying upgrader or not. This pr make it available at loading stage.
2. Swap the order of parsing instruction and operator, because instruction needs to know the operator first because deciding whether applying upgrader or not (change `OP` to `CALL` or not).

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516/)

[ghstack-poisoned]

Commit:
Update on "[Operator Versioning][Edge] Have operator version number available at the loading stage"


1. operator version is needed to decide whether applying upgrader or not. This pr make it available at loading stage.
2. Swap the order of parsing instruction and operator, because instruction needs to know the operator first because deciding whether applying upgrader or not (change `OP` to `CALL` or not).

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516/)

[ghstack-poisoned]

Commit:
Update on "[Operator Versioning][Edge] Have operator version number available at the loading stage"




1. operator version is needed to decide whether applying upgrader or not. This pr make it available at loading stage.
2. Swap the order of parsing instruction and operator, because instruction needs to know the operator first because deciding whether applying upgrader or not (change `OP` to `CALL` or not).

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516/)

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516)

[ghstack-poisoned]

iseeyuan commented on torch/csrc/jit/mobile/import.cpp:
```diff
@@ -375,18 +380,26 @@ void BytecodeDeserializer::parseMethods(
         function.get());
 
+    // 2. Decides if upgrader is needed
+    bool use_upgrader =
```
Is it used?

iseeyuan replied:
It's used in #67731. Would be great to move it there for self-contain.

cccclai replied:
`bool use_upgrader` is a flag and it's decided by the loaded model operator version.. This variable is initialized but not used. One most important change in this pr is to swapping the sequence. Previously `parseInstructions` -> `parseOperators`, now `parseOperators` -> `parseInstructions`.

Commit:
Update on "[Operator Versioning][Edge] Have operator version number available at the loading stage"




1. operator version is needed to decide whether applying upgrader or not. This pr make it available at loading stage.
2. Swap the order of parsing instruction and operator, because instruction needs to know the operator first because deciding whether applying upgrader or not (change `OP` to `CALL` or not).

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516/)

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516)

[ghstack-poisoned]

Commit:
Update on "[Operator Versioning][Edge] Have operator version number available at the loading stage"




1. operator version is needed to decide whether applying upgrader or not. This pr make it available at loading stage.
2. Swap the order of parsing instruction and operator, because instruction needs to know the operator first because deciding whether applying upgrader or not (change `OP` to `CALL` or not).

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516/)

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516)

[ghstack-poisoned]

Commit:
Update on "[Operator Versioning][Edge] Have operator version number available at the loading stage"




1. operator version is needed to decide whether applying upgrader or not. This pr make it available at loading stage.
2. Swap the order of parsing instruction and operator, because instruction needs to know the operator first because deciding whether applying upgrader or not (change `OP` to `CALL` or not).

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516/)

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516)

[ghstack-poisoned]

Commit:
Update on "[Operator Versioning][Edge] Have operator version number available at the loading stage"




1. operator version is needed to decide whether applying upgrader or not. This pr make it available at loading stage.
2. Swap the order of parsing instruction and operator, because instruction needs to know the operator first because deciding whether applying upgrader or not (change `OP` to `CALL` or not).

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516/)

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516)

[ghstack-poisoned]

cccclai commented on torch/csrc/jit/mobile/import.cpp:
```diff
@@ -376,18 +381,26 @@ void BytecodeDeserializer::parseMethods(
+    // 3. If upgrader is needed, change change the OP instrunction to CALL
+    // instruction (In next PR, use_upgrader will be parsed to parseInstruction
+    // function and do the actual change)
+    parseInstructions(
```
Address in the latest commit. `upgrader_mobile.h` will register as function directly, and if upgrader is needed, the instruction will be swapped with CALL operator and it's according function.

iseeyuan reviewed (approved):
LGTM

Commit:
Update on "[Operator Versioning][Edge] Have operator version number available at the loading stage"




1. operator version is needed to decide whether applying upgrader or not. This pr make it available at loading stage.
2. Swap the order of parsing instruction and operator, because instruction needs to know the operator first because deciding whether applying upgrader or not (change `OP` to `CALL` or not).

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516/)

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516)

[ghstack-poisoned]

Commit:
Update on "[Operator Versioning][Edge] Have operator version number available at the loading stage"




1. operator version is needed to decide whether applying upgrader or not. This pr make it available at loading stage.
2. Swap the order of parsing instruction and operator, because instruction needs to know the operator first because deciding whether applying upgrader or not (change `OP` to `CALL` or not).

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516/)

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516)

[ghstack-poisoned]

Commit:
Update on "[Operator Versioning][Edge] Have operator version number available at the loading stage"




1. operator version is needed to decide whether applying upgrader or not. This pr make it available at loading stage.
2. Swap the order of parsing instruction and operator, because instruction needs to know the operator first because deciding whether applying upgrader or not (change `OP` to `CALL` or not).

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516/)

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516)

[ghstack-poisoned]
